Upgrade helper for bare native projects (#24773)
expo/expo: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx

description: Learn how to incrementally upgrade the Expo SDK version in your project.
---

import { Terminal } from '~/ui/components/Snippet';
import { Step } from '~/ui/components/Step';
import { Tabs, Tab } from '~/ui/components/Tabs';

> **info** We recommend upgrading SDK versions incrementally, one at a time. Doing so will help you pinpoint breakages and issues that arise during the upgrade process.

> Expo maintains ~6 months of backward compatibility. Once an SDK version has been deprecated, you will no longer be able to use the Expo Go app. However, you will still be able to publish updates via `eas update`, and build using `eas build`. Deprecations **will not** affect the standalone apps you have in production or your [development builds](/develop/development-builds/introduction/).

Reviewer: ```suggestion
Expo maintains ~6 months of backward compatibility. Once an SDK version has been deprecated, you will no longer be able to use it in the Expo Go app. SDK deprecations **will not** affect the standalone apps you have in production or your [development builds](/develop/development-builds/introduction/), and backwards compatibility for older SDK versions on EAS services tends to be much longer, but not forever.
```

## How to upgrade to the latest SDK version

<Step label="1">

### Upgrade EAS CLI

Update to the latest version of EAS CLI (if you use it):

<Terminal cmd={['$ npm i -g eas-cli']} />

> </Step>

Reviewer: let's remove this EAS CLI step

<Step label="2">

### Upgrade the Expo SDK

Install the new version of the Expo package:

<Tabs>
<Tab label="npm">

> <Terminal cmd={['$ npm install expo@latest']} />

Reviewer: ```suggestion
<Terminal cmd={['$ npm install expo@latest']} />
```

should we show two options here?

```
# Install latest
npm install expo@latest

# Install a specific SDK version (eg: SDK 49)
npm install expo@49
```

</Tab>

<Tab label="yarn">

<Terminal cmd={['$ yarn install expo@latest']} />

</Tab>
</Tabs>

</Step>

<Step label="3">

### Upgrade dependencies

> Upgrade all dependencies to match SDK 49:

Reviewer: ```suggestion
Upgrade all dependencies to match the installed SDK version.
```

</Step>

<Step label="4">

### Follow the release notes for any other instructions

Read the [SDK changelogs](#sdk-changelogs) for the SDK version you are upgrading to. They contain important information about breaking changes, deprecations, and other changes that may affect your app.

</Step>

## Upgrading native projects
If you manage your own native projects (bare workflow), you will need to manually apply any changes to your native Android and iOS project folders. Our [bare upgrade advisor](/bare/upgrade) will show you a diff that describes the changes you need to make. Alternatively, you could consider [adopting prebuild](/guides/adopting-prebuild) for easier upgrades in the future.

## SDK Changelogs
Each SDK announcement blog post contains deprecations, breaking changes, and anything else that might be unique to that particular SDK version. When upgrading, be sure to check these out to make sure you don't miss anything.

### SDK 49

[Blog Post](https://blog.expo.dev/expo-sdk-49-c6d398cdf740)

### SDK 48

[Blog Post](https://blog.expo.dev/expo-sdk-48-ccb8302e231)

### SDK 47

[Blog Post](https://blog.expo.dev/expo-sdk-47-a0f6f5c038af)

> ### SDK 46 [DEPRECATED]

Reviewer: can we move all deprecated changelogs under a heading called Deprecated SDK Versions and then remove the [DEPRECATED] text from the title and change to a heading such as `####` that won't be included in the sidebar?

description: Learn how to incrementally upgrade the Expo SDK version in your project.
---

import { Terminal } from '~/ui/components/Snippet';
import { Step } from '~/ui/components/Step';
import { Tabs, Tab } from '~/ui/components/Tabs';

> **info** We recommend upgrading SDK versions incrementally, one at a time. Doing so will help you pinpoint breakages and issues that arise during the upgrade process.

> Expo maintains ~6 months of backward compatibility. Once an SDK version has been deprecated, you will no longer be able to use it in the Expo Go app. SDK deprecations **will not** affect the standalone apps you have in production or your [development builds](/develop/development-builds/introduction/), and backwards compatibility for older SDK versions on EAS services tends to be much longer, but not forever.

Reviewer 2: ```suggestion
Expo maintains approximately 6 months of backward compatibility. Once an SDK version has been deprecated, you can no longer use it in the Expo Go app. SDK deprecations **will not** affect the standalone apps you have in production or your [development builds](/develop/development-builds/introduction/), and backwards compatibility for older SDK versions on EAS services tends to be much longer, but not forever.
```

- First change is unrelated to this PR but we should word out instead of using a tilde. Not good for screen readers, etc.

## How to upgrade to the latest SDK version

<Step label="1">

### Upgrade the Expo SDK

Install the new version of the Expo package:

<Tabs>
<Tab label="npm">

<Terminal
  cmd={[
    '# Install latest',

>     '',

Reviewer 2: ```suggestion
```

    '$ npm install expo@latest',
    '',

>     '# Install a specific SDK version (eg: SDK 49)',

Reviewer 2: ```suggestion
    '# Install a specific SDK version (for example, SDK 49)',
```

    'npm install expo@49',
  ]}
/>

</Tab>

> <Tab label="yarn">

Reviewer 2: ```suggestion
<Tab label="Yarn">
```

Branding case convention as per our writing style guide.

<Terminal
  cmd={[
    '# Install latest',

>     '',

Reviewer 2: ```suggestion
```

    '$ yarn add expo@latest',
    '',

>     '# Install a specific SDK version (eg: SDK 49)',

Reviewer 2: ```suggestion
    '# Install a specific SDK version (for example, SDK 49)',
```

    'yarn add expo@49',
  ]}
/>

</Tab>
</Tabs>

</Step>

<Step label="2">

### Upgrade dependencies

Upgrade all dependencies to match the installed SDK version.

<Terminal cmd={['$ npx expo install --fix']} />

</Step>

<Step label="3">

### Follow the release notes for any other instructions

Read the [SDK changelogs](#sdk-changelogs) for the SDK version you are upgrading to. They contain important information about breaking changes, deprecations, and other changes that may affect your app.

</Step>

> ## Upgrading native projects

Reviewer 2: ```suggestion
## Upgrade native projects
```

> If you manage your own native projects (bare workflow), you will need to manually apply any changes to your native Android and iOS project folders. Our [bare upgrade advisor](/bare/upgrade) will show you a diff that describes the changes you need to make. Alternatively, you could consider [adopting prebuild](/guides/adopting-prebuild) for easier upgrades in the future.

Reviewer 2: ```suggestion
If you manage your own native projects (bare workflow), you will need to manually apply any changes to your native **android** and **ios** project directories. Our [bare upgrade advisor](/bare/upgrade) will show you a diff that describes the changes you need to make. Alternatively, you could consider [adopting prebuild](/guides/adopting-prebuild) for easier upgrades in the future.
```

## SDK Changelogs

Each SDK announcement blog post contains deprecations, breaking changes, and anything else that might be unique to that particular SDK version. When upgrading, be sure to check these out to make sure you don't miss anything.

### SDK 49

[Blog Post](https://blog.expo.dev/expo-sdk-49-c6d398cdf740)

### SDK 48

[Blog Post](https://blog.expo.dev/expo-sdk-48-ccb8302e231)

### SDK 47

[Blog Post](https://blog.expo.dev/expo-sdk-47-a0f6f5c038af)

> ## Deprecated SDK Version Changelogs

Reviewer 2: ```suggestion
### Deprecated SDK Version Changelogs
```

IMO, this should be under SDK changelogs as a sub section.

Each SDK announcement blog post contains deprecations, breaking changes, and anything else that might be unique to that particular SDK version. When upgrading, be sure to check these out to make sure you don't miss anything.

> ### SDK 50 (alpha)

Reviewer: heads up - we're using the name "canary" to refer to our continuous pre-releases prior to beta

Author: That showed up here from fixing conflicts from main, which references SDK 50 alpha here. I just updated it to `(canary)`
